cloudinit/config/cc_mounts.py

```diff
@@ -223,6 +223,63 @@ def suggested_swapsize(memsize=None, maxsize=None, fsys=None):
     return size
 
 
+def get_fstype(path):
+    """Return the type of the filesystem that path is on.
+    If path is a mount point itself, then the type of the filesystem mounted
+    on path rather than its parent dir."""
+    return util.get_mount_info(path)[1]
+
+
+def create_swapfile(fname, size):
+    """Size is in MB."""
+
+    def unlink_f(fname):
+        if os.path.exists(fname):
+            os.unlink(fname)
+
+    def create_swap(fname, size, method):
```

Comment:
Do these two functions need to be nested?

Reply:
I don't think they specifically need to be nested, but I don't see other usage for create_swap() other than inside this function. It's a very local routine that is called three times. I'm gonna keep this here for now.